modules/openapi-generator/src/main/resources/python/configuration.mustache

```diff
@@ -301,6 +400,29 @@ class Configuration(object):
 {{/isOAuth}}
 {{/authMethods}}
         }
+{{#hasHttpSignatureMethods}}
+
+    def load_private_key(self):
+        """Load the private key used to sign HTTP requests.
+           The private key is used to sign HTTP requests as defined in
+           https://datatracker.ietf.org/doc/draft-cavage-http-signatures/.
+        """
+        if self.private_key is not None:
+            return
+        with open(self.private_key_path, "rb") as f:
+            # Decode PEM file and determine key type from PEM header.
+            # Supported values are "RSA PRIVATE KEY" and "ECDSA PRIVATE KEY".
+            keys = pem.parse(f.read())
+            if len(keys) != 1:
+                raise Exception("File must contain exactly one private key")
+            key = keys[0].as_text()
+            if key.startswith("-----BEGIN RSA PRIVATE KEY-----"):
+                self.private_key = RSA.importKey(key)
+            elif key.startswith("-----BEGIN EC PRIVATE KEY-----"):
+                self.private_key = ECC.importKey(key)
+            else:
+                raise Exception("Unsupported key")
+{{/hasHttpSignatureMethods}}
```

Comment:
If we create the signing.py module, it would also be helpful if we could move this function into it
That way all of our signing code would be in one place
Again we would need to conditionally include the `from petstore_api import signing` import up top

Reply:
sure. I'm starting to move the code to a signing.mustache.

Reply:
What is the right way to determine if hasHttpSignatureMethods is set in the Java code?

Reply:
You would loop over fullAuthMethods or authMethods and if any of them have item. isHttpSignature set to True then you should add the signing.mustache template.